### PR TITLE
OOP school library: Set up associations

### PR DESCRIPTION
**Pull request by gitbygaurav**
Hii CR 👋

WelcomeGIF

In this PR, I have done the following changes:-

- [x] I have created the Classroom class that will have the has-many/belongs-to relationship with Student
- [x] I have created the Book and Rental classes.
- [x] I have create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental.

Please review my changes,
Thanks 🍻

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -25,4 +25,8 @@ def can_use_services?
   def of_age?
     @age < 18
   end
+
```
Comment (hiromataba) [inline person.rb:29]: - [ ] *@rentals should be initialized. Link to the example about constructors in a previous [lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars) 
    @rentals = []

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi @gauravgangwarweb    :wave: :wave: 

almost-there

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Your PR has a proper title and description  :+1: 
- Your linters are passing :heavy_check_mark:  
- Profesional README file :rocket: 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (gitbygaurav) [reply person.rb:29]: Hii @hiromataba ,
thanks for suggesting changes, I have implemented all the required changes.

File: student.rb [hunk 2]
```diff
@@ -9,4 +9,9 @@ def initialize(classroom, age, name = 'Unknown', parent_permission: true)
   def play_hockey
     '¯(ツ)/¯'
   end
+
```
Comment (BenMukebo) [inline student.rb:12]: - [OPTIONAL] Check for a getter method for `@classroom`. Kindly let's add this : `attr_reader :classroom` ✍🏻 
- Note: Getter methods help you retrieve the value of an instance variable. If you don’t have a getter method, trying to retrieve a variable is going to result in an error.

**Review by BenMukebo — APPROVED** [review]
### Hi @gauravgangwarweb, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 



### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [OPTIONAL] When I click the link to the `MIT.md` file that you added, I get this 404 `error` Please update your Readme, it will be nice to  update this section(`License`) with the type of License and a link to the license file, as this will make your readme more informative and professional. 👍🏻 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**